compiler/rustc_middle/src/dep_graph/mod.rs

```diff
@@ -28,8 +30,12 @@
 //!   could not be instantiated because the current compilation session
 //!   contained no `DefId` for thing that had been removed.
 //!
+//! ## `DepNode` creation and "inference"
+//!
+// Where is the `define_dep_nodes!()` macro?
+// How much of this section is outdated?
 //! `DepNode` definition happens in the `define_dep_nodes!()` macro. This macro
-//! defines the `DepKind` enum and a corresponding `DepConstructor` enum. The
+//! defines the `DepKind` enum and a corresponding [`DepConstructor`] enum. The
```

Comment:
Drive-by: DepConstructor should really just be a `mod`, since all it does is provide a namespace for functions constructing `DepNode`s.